### PR TITLE
Add bed levelling documentation and helper tools

### PR DESCRIPTION
**Pull request by KevinOConnor**
This series is an attempt to improve the bed levelling documentation in Klipper.  It also adds a few new helper tools (PROBE_CALIBRATE, Z_ENDSTOP_CALIBRATE, and MANUAL_PROBE). The main starting point for the new documentation is: https://github.com/KevinOConnor/klipper/blob/work-bed-level-20190206/docs/Bed_Level.md

The goal of this series is to make it easier for new users to improve the bed levelling process.

@mcmatrix - this series will likely impact the menu work that you've been doing.

@Arksine - I think it would be good to add a document on mesh bed levelling.  I was thinking of trying to tackle that next.  If you have a list of important things to note in such a document then let me know.

Comments?
-Kevin

### CODE REVIEW

File: docs/Probe_Calibrate.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+This document describes the method for calibrating the x, y, and z
+offsets of an "automatic z probe" in Klipper. This is useful for users
+that have a `[probe]` or `[bltouch]` section in their config file.
+
+# Calibrating probe X and Y offsets
+
+To calibrate the X and Y offset, navigate to the OctoPrint "Control"
```
Comment (trevjonez) [inline docs/Probe_Calibrate.md]: should it be issue G1 commands rather than or in addition to octoprint control?

**Comment by Arksine** [comment]
@KevinOConnor I'll come up with some details on bed_mesh and get them to you.    Fade is easily the feature that is most difficult for new users to understand, which is why I disabled it by default.  I also think it would be useful to detail how point generation works.

File: docs/Bed_Level.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+half way between two positions. For example, if one received the
+following prompt from a TESTZ command:
+```
+Recv: // Z position: 0.130 --> 0.230 <-- 0.280
+```
+Then a `TESTZ Z=-` would move the nozzle to a Z position of 0.180
+(half way between 0.130 and 0.230). One can use this feature to help
+rapidly narrow down to a consistent friction. It is also possible to
+use `Z=++` and `Z=--` to return directly to a past measurement - for
+example, after the above prompt a `TEST Z=--` command would move the
```
Comment (Arksine) [inline docs/Bed_Level.md]: I think this should be `TESTZ Z=--`

Comment (KevinOConnor) [reply docs/Bed_Level.md]: Oops, thanks.

File: docs/Probe_Calibrate.md (hunk 1 above)
Comment (KevinOConnor) [reply docs/Probe_Calibrate.md]: Well, the high level idea is to ask the user to get the nozzle near the center of the bed.  If you think there's a better way to word that then let me know.

**Comment by KevinOConnor** [comment]
FYI, I've merged this into the master branch.

-Kevin

**Pull request closed without merging**